File: basaltpass-backend/internal/model/order.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// OrderStatus 订单状态枚举
type OrderStatus string

const (
	OrderStatusPending   OrderStatus = "pending"   // 待支付
	OrderStatusPaid      OrderStatus = "paid"      // 已支付
	OrderStatusExpired   OrderStatus = "expired"   // 已过期
	OrderStatusCancelled OrderStatus = "cancelled" // 已取消
)

// Order 订单模型
type Order struct {
	gorm.Model
	OrderNumber      string      `gorm:"uniqueIndex;size:32;not null"` // 订单号
	UserID           uint        `gorm:"not null;index"`               // 用户ID
	PriceID          uint        `gorm:"not null;index"`               // 价格ID
	CouponID         *uint       `gorm:"index"`                        // 优惠券ID（可选）
	Status           OrderStatus `gorm:"size:20;not null;index"`       // 订单状态
	Quantity         float64     `gorm:"not null;default:1"`           // 数量
	BaseAmount       int64       `gorm:"not null"`                     // 基础金额（分）
	DiscountAmount   int64       `gorm:"not null;default:0"`           // 折扣金额（分）
	TotalAmount      int64       `gorm:"not null"`                     // 总金额（分）
	Currency         string      `gorm:"size:3;not null"`              // 币种
	Description      string      `gorm:"size:500"`                     // 订单描述
	ExpiresAt        time.Time   `gorm:"not null;index"`               // 过期时间
	PaidAt           *time.Time  // 支付时间
	SubscriptionID   *uint       `gorm:"index"`     // 关联的订阅ID
	PaymentSessionID *uint       `gorm:"index"`     // 关联的支付会话ID
	Metadata         JSONB       `gorm:"type:json"` // 元数据

	// 关联
	User           User            `gorm:"foreignKey:UserID"`
	Price          Price           `gorm:"foreignKey:PriceID"`
	Coupon         *Coupon         `gorm:"foreignKey:CouponID"`
	Subscription   *Subscription   `gorm:"foreignKey:SubscriptionID"`
	PaymentSession *PaymentSession `gorm:"foreignKey:PaymentSessionID"`
}

// TableName 指定表名
func (Order) TableName() string {
	return "market_orders"
}

File: basaltpass-frontend/src/shared/api/subscription/payment/order.ts
import client from '../../client';

// orderstatustranslated
export type OrderStatus = 'pending' | 'paid' | 'expired' | 'cancelled';

// createorderrequest
export interface CreateOrderRequest {
  user_id: number;
  price_id: number;
  quantity?: number;
  coupon_code?: string;
}

// orderresponse
export interface OrderResponse {
  id: number;
  order_number: string;
  user_id: number;
  price_id: number;
  coupon_id?: number;
  status: OrderStatus;
  quantity: number;
  base_amount: number;
  discount_amount: number;
  total_amount: number;
  currency: string;
  description: string;
  expires_at: string;
  paid_at?: string;
  created_at: string;
  updated_at: string;
  
  // translated
  price?: {
    id: number;
    amount_cents: number;
    currency: string;
    billing_interval: string;
    billing_interval_count: number;
    plan: {
      id: number;
      display_name: string;
      product: {
        id: number;
        name: string;
        description: string;
      };
    };
  };
  coupon?: {
    id: number;
    code: string;
    discount_type: string;
    discount_value: number;
  };
}

// orderAPItranslated
export class OrderAPI {
  // createorder
  async createOrder(data: CreateOrderRequest): Promise<OrderResponse> {
    const response = await client.post('/api/v1/orders', data);
    return response.data.data;
  }

  // getorderdetails
  async getOrder(orderId: number, options?: { activate?: boolean }): Promise<OrderResponse> {
    const params = options?.activate ? { activate: 1 } : undefined;
    const response = await client.get(`/api/v1/orders/${orderId}`, { params });
    return response.data.data;
  }

  // translatedordertranslatedgetorder
  async getOrderByNumber(orderNumber: string): Promise<OrderResponse> {
    const response = await client.get(`/api/v1/orders/number/${orderNumber}`);
    return response.data.data;
  }

  // getuserorderlist
  async listOrders(limit?: number): Promise<OrderResponse[]> {
    const params = limit ? { limit } : {};
    const response = await client.get('/api/v1/orders', { params });
    return response.data.data;
  }
}

// translatedAPItranslated
export const orderAPI = new OrderAPI();

// translated
export const createOrder = (data: CreateOrderRequest) => orderAPI.createOrder(data);
export const getOrder = (orderId: number, options?: { activate?: boolean }) => orderAPI.getOrder(orderId, options);
export const getOrderByNumber = (orderNumber: string) => orderAPI.getOrderByNumber(orderNumber);
export const listOrders = (limit?: number) => orderAPI.listOrders(limit); 